sshd-contrib/src/main/java/org/apache/sshd/common/util/security/androidopenssl/AndroidOpenSSLSecurityProviderRegistrar.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.sshd.common.util.security.androidopenssl;
+
+import java.security.Provider;
+import java.security.Security;
+
+import org.apache.sshd.common.util.security.AbstractSecurityProviderRegistrar;
+
+public class AndroidOpenSSLSecurityProviderRegistrar extends AbstractSecurityProviderRegistrar {
+    public AndroidOpenSSLSecurityProviderRegistrar() {
+        super("AndroidOpenSSL");
+    }
+
+    @Override
+    public boolean isSupported() {
+        return true;
+    }
```

Comment:
I would rather have `isSupported` check something that really shows in the security provider is supported - e.g., the existence of some class, system property, etc... - something that would prevent it from being "supported" if it is not running in an Android environment.

Reply:
In fact there is such a system property. I've changed it to check for that. Thanks for the suggestion.